arcade/gui/widgets/__init__.py

```diff
@@ -167,6 +178,20 @@ def on_update(self, dt):
         """Custom logic which will be triggered."""
         pass
 
+    def _event_pos_relative_to_self(self, mouse_event: UIMouseEvent) -> Vec2 | None:
+        """Gets coords relative to bottom left if inside the widget.
+
+        Args:
+            mouse_event: Any :py:class:`UIMouseEvent`.
+        Returns:
+            ``None`` if outside the widget or coords relative to
+             the widget's bottom left.
+        """
+        pos = mouse_event.pos
+        if not self.rect.point_in_rect(pos):
+            return None
+        return Vec2(pos[0] - self.left, pos[1] - self.bottom)
+
```

Comment:
Update to use the new `Rect` method.

Reply:
I didn't update it yet because I want to give it more thought after rereading the UI code and considering the following:
1. 3.9 features will be our new minimum
2. The new rect method seems to perform more ops
3. It may add even more call overhead